src/Algebra/Properties/CancellativeCommutativeSemiring.agda

```diff
@@ -0,0 +1,45 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties of operations in CancellativeCommutativeSemiring.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (CancellativeCommutativeSemiring)
+open import Algebra.Definitions using (AlmostRightCancellative)
+open import Data.Sum.Base using (_⊎_; inj₁; inj₂)
+open import Function using (_$_)
+open import Relation.Binary using (Decidable)
+import Relation.Binary.Reasoning.Setoid as EqReasoning
+open import Relation.Nullary using (yes; no)
+open import Relation.Nullary.Negation using (contradiction)
+
+module Algebra.Properties.CancellativeCommutativeSemiring
+  {a ℓ} (R : CancellativeCommutativeSemiring a ℓ)
+  (open CancellativeCommutativeSemiring R) (_≟_ : Decidable _≈_) where
+
+open EqReasoning setoid
+
+*-cancelʳ-nonZero : AlmostRightCancellative _≈_ 0# _*_
```

Comment:
A generalisation of this lemma (and its opposite lemma) should be in `Algebra.Consequences.Setoid`. The generalisation should then be used here.

Reply:
Done.